### PR TITLE
[PodXCConfig] Use recursive dependent targets

### LINKED ISSUE #5264: Framework Search Path is missing dependent targets when using `use_framework!` and subspecs
🌈

* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

I did a [sample code](http://felipecypriano.com/files/search-path-fail.zip) matching my real world problem when I tried to migrate to `use_frameworks!`, no idea why but the problem is that the Framework Search Path for some pods don't include all the dependencies, specifically the ones that come from another pod.

I think this is related to https://github.com/CocoaPods/CocoaPods/issues/5241. I didn't add the sample code there because I wasn't sure.

## What did you do?

1. Download the [sample project](http://felipecypriano.com/files/search-path-fail.zip)
2. Run `pod install` in the `app` directory
3. Open the workspace
4. Try to build the App target (or each pod target directly if you want to double check something)

## What did you expected to happen?

The build to work.

## What happened instead?

Build fails to build the `asserts` pod:

> search-path-fail/logger/Classes/Logger.h:1:9: 'CocoaLumberjack/CocoaLumberjack.h' file not found
> search-path-fail/asserts/Classes/Asserts.m:1:9: Could not build module 'logger'

This happens because CocoaLumberjack (which is needed to build `logger` that asserts depend on) is not added to `asserts`' Framework Search Path.

Adding the missing dependency to Framework Search Path to both `asserts`  and `sharedlib-Core` makes the build successful.

## Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'

platform :ios, '8.0'
use_frameworks!

target 'App' do

  pod 'logger', path: '../logger'
  pod 'sharedlib/Core', path: '../sharedlib'
  pod 'asserts', path: '../asserts'

  target 'Test' do
    inherit! :search_paths

    pod 'testkit', path: '../testkit'
    pod 'sharedlib/Testing', path: '../sharedlib'
  end
end
```

## Podspecs

Unnecessary metadata ommited.

### logger

```
Pod::Spec.new do |s|
  s.name         = "logger"
  s.default_subspecs = 'Core'

  s.subspec 'Core' do |c|
    c.source_files  = "Classes/**/*.{h,m}"

    c.dependency "CocoaLumberjack"
  end

  s.subspec 'Crashlytics' do |cr|
    cr.dependency 'CrashlyticsRecorder'
  end
end
```

### asserts

```
Pod::Spec.new do |s|
  s.name         = "asserts"
  s.source_files  = "Classes", "Classes/**/*.{h,m}"
  s.exclude_files = "Classes/Exclude"

  s.dependency "logger"
end
```

### sharedlib

```
Pod::Spec.new do |s|
  s.name         = "sharedlib"
  s.subspec 'Core' do |core|
    core.source_files  = "Classes/Core/*.{h,m}"
    core.dependency "asserts"
    core.dependency "logger"
  end

  s.subspec 'Testing' do |t|
    t.source_files  = "Classes/Testing/*.{h,m}"
    t.dependency 'testkit'
  end
end
```

### testkit

```
Pod::Spec.new do |s|
  s.name         = "testkit"
  s.source_files  = "Classes/**/*.{h,m}"

  s.framework = 'XCTest'

  s.dependency 'asserts'
end
```

### PR DESCRIPTION
**Pull request by segiddins**
Closes #5264 

- [x] CHANGELOG

### CODE REVIEW

**Comment by neonichu** [comment]
LGTM :+1:

File: lib/cocoapods/target/pod_target.rb [hunk 1]
```diff
@@ -184,6 +184,15 @@ def dependencies
       end.uniq
     end
 
+    # @return [Array<PodTarget>] the recursive targets that this target has a
+    #         dependency upon.
+    #
+    def recursive_dependent_targets
+      targets = dependent_targets + dependent_targets.flat_map(&:dependent_targets)
```
Comment (AliSoftware) [inline lib/cocoapods/target/pod_target.rb]: Doesn't this stop the recursion at one level only? I'd have though the right recursive implementation should be `dependent_targets + dependent_targets.flat_map(&:recursive_dependent_targets)`?

**Comment by segiddins** [comment]
@AliSoftware thank you, good catch!

**Comment by fcy** [comment]
FWIW the fix worked on my real project (the base for #5264) 👏 and thank you!!!

**Pull request merged**